server/utils/queryHelpers/pubGet.ts

```diff
@@ -16,22 +17,37 @@ const resolveGetPubWhereQuery = (where: GetPubWhere) => {
 	return { id: where.id };
 };
 
+const getFacetsForPub = async (options: PubGetOptions, where: GetPubWhere) => {
+	if (options.getFacets) {
+		const pubId =
+			'id' in where
+				? where.id
+				: (await Pub.findOne({ where, attributes: ['slug', 'communityId', 'id'] })).id;
+		return { facets: await fetchFacetsForScope({ kind: 'pub', id: pubId }) };
+	}
+	return null;
+};
+
 export const getPub = async (where: GetPubWhere, options: PubGetOptions = {}) => {
-	const pubData = await Pub.findOne({
-		where: resolveGetPubWhereQuery(where),
-		...buildPubOptions({
-			getMembers: true,
-			getCollections: true,
-			getExports: true,
-			getEdges: 'approved-only',
-			...options,
+	where = resolveGetPubWhereQuery(where);
+	const [pubData, facets] = await Promise.all([
```

Comment:
interesting, yeah i see why it could be the default way to get facets off scopes(though i like property instead if facet. lol sorry for the late contrib). 

can you explain how TS might find a facet for, say `CitationStyleKind, and infer that to be the same `CitationStyleKind` type for `pub`

Reply:
or is this just what zod does

Reply:
Sorry to be 15 days late to this — the key is that we're getting rid of the `pub.citationStyle` property, so nothing has to do any matching to Pub properties. They're disappearing from the Pub per se and reappearing on facets attached to the Pub.